referenceframe/urdf.go

```diff
@@ -0,0 +1,299 @@
+			}
+
+			mc.Joints = append(mc.Joints, thisJoint)
+
+			// Generate child link translation and orientation data, which is held by this joint per the URDF design
+			childXYZ := convStringAttrToFloats(jointElem.Origin.XYZ)
+			childRPY := convStringAttrToFloats(jointElem.Origin.RPY)
+			childEA := spatial.EulerAngles{Roll: childRPY[0], Pitch: childRPY[1], Yaw: childRPY[2]}
+			childOrient, err := spatial.NewOrientationConfig(childEA.EulerAngles())
+
+			// Note the conversion from meters to mm
+			childLink.Translation = spatial.TranslationConfig{childXYZ[0] * 1000, childXYZ[1] * 1000, childXYZ[2] * 1000}
+			childLink.Orientation = *childOrient
+
+			if err != nil {
+				return nil, err
+			}
+		case "fixed":
+			// Handle fixed joint -> static link conversion instead of adding to Joints[]
+			thisLink := JsonLink{ID: jointElem.Name, Parent: jointElem.Parent.Link}
+
+			linkXYZ := convStringAttrToFloats(jointElem.Origin.XYZ)
+			linkRPY := convStringAttrToFloats(jointElem.Origin.RPY)
+			linkEA := spatial.EulerAngles{Roll: linkRPY[0], Pitch: linkRPY[1], Yaw: linkRPY[2]}
+			linkOrient, err := spatial.NewOrientationConfig(linkEA.AxisAngles())
+
+			// Note the conversion from meters to mm
+			thisLink.Translation = spatial.TranslationConfig{linkXYZ[0] * 1000, linkXYZ[1] * 1000, linkXYZ[2] * 1000}
+			thisLink.Orientation = *linkOrient
+
+			if err != nil {
+				return nil, err
+			}
+
+			mc.Links = append(mc.Links, thisLink)
+		default:
+			return nil, NewUnsupportedJointTypeError(jointElem.Type)
+		}
+
+		if err != nil {
+			return nil, err
+		}
+
+		mc.Links = append(mc.Links, childLink)
+	}
+
+	// Handle links
+	for _, linkElem := range urdf.Links {
+		// Skip any world links
+		if linkElem.Name == "world" {
+			continue
+		}
+
+		// Find matching links which already exist, take care of geometry if collision elements are detected
+		hasCollision := false
+		if len(linkElem.Collision) > 0 {
+			hasCollision = true
+		}
+
+		for idx, prefabLink := range mc.Links {
+			if prefabLink.ID == linkElem.Name && hasCollision {
+				geoCfg, _ := createConfigFromCollision(linkElem)
+				mc.Links[idx].Geometry = geoCfg
+				break
+			}
+		}
+
+		// In the event the link does not already exist in the ModelConfig, we will have to generate it now
+		// Most likely, this is a link normally whose parent is the World
+		if _, ok := parentMap[linkElem.Name]; !ok {
+			thisLink := JsonLink{ID: linkElem.Name, Parent: World}
+
+			linkEA := spatial.EulerAngles{Roll: 0.0, Pitch: 0.0, Yaw: 0.0}
+			linkOrient, err := spatial.NewOrientationConfig(linkEA.AxisAngles())
+
+			thisLink.Translation = spatial.TranslationConfig{0.0, 0.0, 0.0}
+			thisLink.Orientation = *linkOrient
+
+			if err != nil {
+				return nil, err
+			}
+
+			if hasCollision {
+				geoCfg, _ := createConfigFromCollision(linkElem)
+				thisLink.Geometry = geoCfg
+			}
+
+			mc.Links = append(mc.Links, thisLink)
+		}
+	}
+
+	return mc.ParseConfig(modelName)
```

Comment:
Would it make sense to return a `ModelConfig` here and wrap in a function that returns a `Model`? That would allow us to write unit tests that check the expected `ModelConfig` directly.

Reply:
Yes! This signature matches a roughly equivalent method over in `model_json.go` as well, so I would be keen to update both files to follow that pattern.

Reply:
I changed this behavior for the URDF code; I broke a test when I attempted to do this in `model_json`, so I paused that and just applied this change to URDF. I would still like to come back and make this behavior common.

Reply:
DATA-877 captures the effort to come back and make this behavior common.